src/transformers/models/clip/modeling_clip.py

```diff
@@ -858,13 +874,36 @@ def forward(
         )
         return_dict = return_dict if return_dict is not None else self.config.use_return_dict
 
-        encoder_states = () if output_hidden_states else None
+        if isinstance(output_hidden_states, list):
+            _output_hidden_states = []
+            for index in output_hidden_states:
+                if index > self.config.num_hidden_layers:
+                    raise ValueError(f"output_hidden_states index {index} is out of range.")
+                elif index < 0:
+                    index = self.config.num_hidden_layers + index + 1
+                    if index > self.config.num_hidden_layers:
+                        raise ValueError(f"output_hidden_states index {index} is out of range.")
+                    _output_hidden_states.append(index)
+                else:
+                    _output_hidden_states.append(index)
+            output_hidden_states = _output_hidden_states
```

Comment:
I recommend being careful with indexes! We probably can have a separate method or function, something like `_prepare_hidden_states_indexes(output_hidden_states, num_layers)`:
1.  In case `output_hidden_states=True` we can return a full list of indexes range(0, num_layers)
2. In case `output_hidden_states=False` we can return an empty list
3. If it's a list
     - we can verify all indexes are in the range [-num_layers, num_layers-1]
     - then, we can convert them to positive integers

It will allow avoid multiple if/else statements later in the code, and I believe will improve readability

In current implementation, there might be an index -100 with num_layers=10, even after `index = self.config.num_hidden_layers + index + 1` it will still be a negative index.

Reply:
Thanks. The check is now a single if statement which validates:

1. Indices are less than self.config.num_hidden_layers
2. Negative indices are not still negative after wrapping
3. Negative indices are less than self.config.num_hidden_layers after wrapping

Then negative indices are wrapped.